server/src/main/java/org/elasticsearch/cluster/metadata/IndexReshardingMetadata.java

```diff
@@ -207,15 +208,22 @@ public static IndexReshardingMetadata newSplitByMultiple(int shardCount, int mul
         return new IndexReshardingMetadata(IndexReshardingState.Split.newSplitByMultiple(shardCount, multiple));
     }
 
-    public IndexReshardingMetadata transitionSplitTargetToHandoff(ShardId shardId) {
+    public static boolean isSplitTarget(ShardId shardId, @Nullable IndexReshardingMetadata reshardingMetadata) {
+        return reshardingMetadata != null && reshardingMetadata.isSplit() && reshardingMetadata.getSplit().isTargetShard(shardId.id());
+    }
+
+    public IndexReshardingMetadata transitionSplitTargetToNewState(
+        ShardId shardId,
+        IndexReshardingState.Split.TargetShardState newTargetState
+    ) {
         assert state instanceof IndexReshardingState.Split;
         IndexReshardingState.Split splitState = (IndexReshardingState.Split) state;
         IndexReshardingState.Split.TargetShardState[] newTargets = Arrays.copyOf(
             splitState.targetShards(),
             splitState.targetShards().length
         );
```

Comment:
Maybe we should make a builder to centralize the logic for creating a new state instead of mutating the existing one?

Reply:
I think you had already created one that I wasn't using. I just switched to that.